### ISSUE TITLE
windows as service does not work

### ISSUE THREAD

**Issue by rmuir** [elastic/elasticsearch#13247]
Trying to follow https://www.elastic.co/guide/en/elasticsearch/reference/current/setup-service-win.html doesn't work with master.

I get this error "The data area passed to a system call is too small"

**Comment by rmuir** [comment]
I think this functionality is also untested.

**Comment by clintongormley** [comment]
@gmarz would you be able to take a look at this?

**Comment by gmarz** [comment]
I get the same thing as well:

```
Exception in thread "main" areSettings(InternalSettingsPreparer.java:87)
	at org.elasticsearch.common.cli.CliTool.<init>(CliTool.java:108)
	at org.elasticsearch.common.cli.CliTool.<init>(CliTool.java:101)
	at org.elasticsearch.bootstrap.BootstrapCLIParser.<init>(BootstrapCLIParser.java:48)
	at org.elasticsearch.bootstrap.Bootstrap.main(Bootstrap.java:226)
	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:28)
The data area passed to a system call is too small.

Failed to start service
```

I'll start digging...

**Comment by nik9000** [comment]
> I think this functionality is also untested.

Very much. I was looking at getting vagrant tests for this too but it'd be very different than the existing bats tests. And I'm not sure what the licenses are for running windows VMs for testing. If someone can point me to something that makes it clear that window's license lets people spin up VMs for testing then I'll dig more into it and try to use the same technique we use for linux. Without bash, I guess.

**Comment by Andy-Peng** [comment]
It is added to the system after the service (win7), run service.bat start, reported the same mistakes.

**Comment by Andy-Peng** [comment]
2015-09-08 11:37:45 Commons Daemon procrun stderr initialized
Java HotSpot(TM) 64-Bit Server VM warning: Using the ParNew young collector with the Serial old collector is deprecated and will likely be removed in a future release
Exception in thread "main" areSettings(InternalSettingsPreparer.java:97)
	at org.elasticsearch.common.cli.CliTool.<init>(CliTool.java:107)
	at org.elasticsearch.common.cli.CliTool.<init>(CliTool.java:100)
	at org.elasticsearch.bootstrap.BootstrapCLIParser.<init>(BootstrapCLIParser.java:48)
	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:221)
	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:35)
Refer to the log for complete error details.

**Comment by Andy-Peng** [comment]
elasticsearch-2.0.0-beta1 , I used the jdk 1.8_25,report this error in win7 service.

2015-09-08 11:37:45 Commons Daemon procrun stderr initialized
Java HotSpot(TM) 64-Bit Server VM warning: Using the ParNew young collector with the Serial old collector is deprecated and will likely be removed in a future release
Exception in thread "main" areSettings(InternalSettingsPreparer.java:97)
	at org.elasticsearch.common.cli.CliTool.<init>(CliTool.java:107)
	at org.elasticsearch.common.cli.CliTool.<init>(CliTool.java:100)
	at org.elasticsearch.bootstrap.BootstrapCLIParser.<init>(BootstrapCLIParser.java:48)
	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:221)
	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:35)
Refer to the log for complete error details.

**Comment by jochen-st** [comment]
I found that changes in elasticsarch.in.bat related to ES_GC_OPTS environment variables caused JAVA_OPTS environment variable to contain "  " two consecutive spaces. These are converted by service.bat into ";;" two consecutive semicolons. This causes procrun to stop adding Java options to service and service start fails with:

    Java HotSpot(TM) 64-Bit Server VM warning: Using the ParNew young collector with the Serial old 
    collector is deprecated and will likely be removed in a future release
    Exception in thread "main" areSettings(InternalSettingsPreparer.java:97)
	at org.elasticsearch.common.cli.CliTool.<init>(CliTool.java:107)
	at org.elasticsearch.common.cli.CliTool.<init>(CliTool.java:100)
	at org.elasticsearch.bootstrap.BootstrapCLIParser.<init>(BootstrapCLIParser.java:48)
	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:221)
	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:35)

However even after manipulating ES_GC_OPTS / JAVA_OPTS to not contain two consecutive spaces (and procrun taking all Java options), service start still fails with

    ERROR: command not specified

**Comment by gmarz** [comment]
@jochen-st yea, that's one of the issues with the script.  The other, which you're running into now has to do with the fact that the new elasticsearch CLI now requires a [start](https://github.com/elastic/elasticsearch/blob/master/distribution/src/main/resources/bin/elasticsearch.bat#L46) argument, which the service doesn't pass.

I'm working on a fix and will open a PR shortly.

**Comment by gmarz** [comment]
@jochen-st sorry, just realized I misread your comment and that you already discovered that the script doesn't pass the `start` param.  I just opened the above PR to address the startup issues.

The stopping issue is a bit more involved.  Apache Commons Daemon requires a static stop method (--StopMethod) in order to shutdown gracefully.  This use to exist (`org.elasticsearch.bootstrap.Elasticsearch.close()`) but was removed in ES 2.x.  The script was never updated, hence the error message.

Not sure the reasoning for removing it and how it should be added back.  I'm going to open a new issue for this.
